src/main/java/be/seeseemelk/mockbukkit/ServerMock.java

```diff
@@ -132,9 +132,38 @@
 	private final MockCommandMap commandMap = new MockCommandMap(this);
 	private final HelpMapMock helpMap = new HelpMapMock();
 
+	private WarningState warningState = WarningState.DEFAULT;
 	private GameMode defaultGameMode = GameMode.SURVIVAL;
 	private ConsoleCommandSender consoleSender;
 
+	//===BEGIN bukkit.yml fields getters support
+	//maybe configurable in the future.
+	//todo: make this section configurable
+	//default values from https://bukkit.fandom.com/wiki/Bukkit.yml
+
+	//settings
+	private boolean allowEnd = true;
+	private String updateFolder = "update";
+	//default worldContainer is current working directory (according to the wiki)
+	private File worldContainer = new File("");
+	private long connectionThrottle = 4000;
+	private String shutdownMessage = "Server closed";
+
+	//spawn-limits
+	private int monsterSpawnLimit = 70;
+	private int animalSpawnLimit = 10;
+	private int waterAnimalSpawnLimit = 15;
+	private int waterAmbientSpawnLimit = 20;
+	private int ambientSpawnLimit = 15	;
+
+	//ticks-per
+	private int ticksPerAnimalSpawns = 400;
+	private int ticksPerMonsterSpawns = 1;
+	private int ticksPerWaterSpawns = 1;
+	private int ticksPerWaterAmbientSpawns = 1;
+	private int ticksPerAmbientSpawns = 1;
+	//===END bukkit.yml fields getters support
```

Comment:
I think having all these fields here pollutes the `ServerMock` class a bit too much.
Ideally these should all have corresponding setters too, in case someone needs to test their plugin under different environments.

With only two fields added I wouldn't have bat an eye but this amount of extra settings needs a cleaner solution in my opinion.
So if you are gonna add all of these settings, then I'd reckon a dedicated `ServerSettingsMock` class would be most appropriate, this would clean up the `ServerMock` class a bit, we could add setters in there and it could also benefit performance since this class is initialized on almost every test, so we could only load the settings into memory as necessary.
The getters in `ServerMock` could then just redirect like this:

```java
public Something getSomething()
{
    return getServerSettings().getSomething();
}
```

And then we would have a  method like this:
```java
public @NotNull ServerSettingsMock getServerSettings()
{
    if (serverSettings = null)
    {
        serverSettings = new ServerSettingsMock(this);
    }

    return serverSettings;
}
```


We could also add existing constants in there, like the MOTD and stuff.